regression_tests/tests/gitbook/test_docs.py

```diff
@@ -0,0 +1,181 @@
+    # Delete the temporary file
+    os.remove(filename)
+    return SnippetExec(snippet, success, output, error)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--docs-folder",
+        dest="docs_folder",
+        default="gitbook",
+        action="store",
+        help="Path to gitbook directory.",
+    )
+
+    parser.add_argument(
+        "--run-subset",
+        dest="run_subset",
+        default=False,
+        action="store_true",
+        help="Run a subset of the gitbook sections.",
+    )
+
+    parser.add_argument(
+        "--sections",
+        dest="sections",
+        default=[],
+        nargs='+',
+        action="store",
+        help="Subset of gitbook sections to run.",
+    )
+
+    args = parser.parse_args()
+
+    # If we do not want to test every code snippet in the documentation, we can `run_subset` instead.
+    if args.run_subset:
+        # The subset of sections we run is specified in `sections_subset`. The for loop expects a list of tuples so we add a dummy 0.
+        args.sections = [(section, 0) for section in args.sections_subset]
+    else:
+        # Otherwise, we look at all files in all sections in the `docs_folder`.
+        args.sections = os.walk(args.docs_folder)
+
+    successfully_ran_all = True
+    for item in args.sections:
+        # Check to see if the section is excluded via `blacklist_sections`.
+        skip = False
+        for blacklist_section in blacklist_sections:
+            current_section = item[0].split("gitbook/")[-1]
+            if current_section == blacklist_section or current_section.startswith(blacklist_section+"/"):
+                skip = True
+                print(">> SKIPPING ", current_section)
+                break
+
+        # skip = False which means we want to run the files in the section.
+        if not skip:
+            for file in glob(os.path.join(item[0], '*.md')):
+                file_name = file.split("gitbook/")[-1]
+
+                # Check to see if the section is excluded via `blacklist_files`.
+                skip = False
+                for blacklist_file in blacklist_files:
+                    if file_name == blacklist_file:
+                        skip = True
+                        print(">> SKIPPING ", file_name)
+
+                # skip = False which means we want to run the file.
+                if not skip:
+                    snippets = get_code(file)
+                    
+                    # Remove any snippet that shows up in `blacklist_snippets`.
+                    if file_name in blacklist_snippets.keys():
+                        snippets = [snippet for snippet in snippets if snippet not in blacklist_snippets[file_name]]
+                    
+                    # If we still have code to run, run it.
+                    if len(snippets) > 0:
+                        snippet_result = run("\n".join(snippets), file.replace(r"/", "_")[:-2] + "py")
```

Comment:
what's this `file.replace(r"/"...)[:-2]` stuff about?

Reply:
That's the name of the temporary file I create and run. I've refactored it so it is a variable named `temp_file_name` for readability.